### PR TITLE
IGAPP-166: flowFixMe in EventsContainer

### PR DESCRIPTION
**Pull request by Taggotty**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

### CODE REVIEW

File: native/src/routes/events/containers/EventsContainer.js [hunk 1]
```diff
@@ -19,7 +19,7 @@ import ErrorCodes from '../../../modules/error/ErrorCodes'
 
 type ContainerPropsType = {|
   path: ?string,
-  events: $ReadOnlyArray<EventModel>,
+  events: ?$ReadOnlyArray<EventModel>,
```
Comment (Taggotty) [inline native/src/routes/events/containers/EventsContainer.js:22]: neccessary for LoadingType

File: native/src/routes/events/containers/EventsContainer.js [hunk 2]
```diff
@@ -90,36 +90,62 @@ const mapStateToProps = (state: StateType, ownProps: OwnPropsType): StatePropsTy
...
+    cities: cities,
+    cityCode: route.city,
+    language: route.language,
+    resourceCache: resourceCache.value,
+    resourceCacheUrl: state.resourceCacheUrl,
+    navigation
+  }
 
-  // $FlowFixMe Flow can't evaluate the status as it is dynamic
+  if (route.status === 'loading') {
```
Comment (Taggotty) [inline native/src/routes/events/containers/EventsContainer.js:138]: This is actually the [recommended solution from the flow config](https://flow.org/en/docs/lang/refinements/). 
To keep the one-liner we would need to redefine the StatusPropsType to have one status looking like the code below. Personally I like the if/else construct more
```javascript
export type StatusPropsType<S: {}, R: {}> =
   { status: 'loading' | 'success' | 'routeNotInitialized' | 'error' } &
  (RouteNotInitializedType
  | LoadingType<$Diff<S, { dispatch: Dispatch<StoreActionType> }>, R>
  | ErrorType<R>
  | LanguageNotAvailableType
  | SuccessType<$Diff<S, { dispatch: Dispatch<StoreActionType> }>, R>)
```

**Review by klinzo — APPROVED** [review]
This does not have a separate issue, does it?

**Comment by Taggotty** [comment]
> This does not have a separate issue, does it?
 
The naming of the issue was quite misleading which is also why I accidentally swapped the names of the branch in IGAPP-197 with IGAPP-166. Naming should now be fixed: see [this issue](https://issues.tuerantuer.org/browse/IGAPP-166) for more info on this PR

**Review by Znippet — APPROVED** [review]

**Review by maxammann — APPROVED** [review]

**Pull request merged**